internal/apis/certmanager/types_certificaterequest.go

```diff
@@ -41,58 +48,88 @@ const (
 	metav1.ListMeta
 
+	// List of CertificateRequests
 	Items []CertificateRequest
 }
 
 // CertificateRequestSpec defines the desired state of CertificateRequest
+//
+// NOTE: It is important to note that the issuer can choose to ignore and/
+// or change any of the requested attributes. How the issuer maps a certificate
+// request to a signed certificate is the full responsibility of the issuer itself.
+// For example, as an edge case, an issuer that inverts the isCA value is
+// free to do so.
 type CertificateRequestSpec struct {
-	// The requested 'duration' (i.e. lifetime) of the Certificate.
-	// This option may be ignored/overridden by some issuer types.
+	// Requested 'duration' (i.e. lifetime) of the Certificate.
 	Duration *metav1.Duration
 
-	// IssuerRef is a reference to the issuer for this CertificateRequest.  If
-	// the `kind` field is not set, or set to `Issuer`, an Issuer resource with
-	// the given name in the same namespace as the CertificateRequest will be
-	// used.  If the `kind` field is set to `ClusterIssuer`, a ClusterIssuer with
-	// the provided name will be used. The `name` field in this stanza is
-	// required at all times. The group field refers to the API group of the
-	// issuer which defaults to `cert-manager.io` if empty.
+	// Reference to the issuer responsible for issuing the certificate.
+	// If the issuer is namespace-scoped, it must be in the same namespace
+	// as the Certificate. If the issuer is cluster-scoped, it can be used
+	// from any namespace.
+	//
+	// The `name` field of the reference must always be specified.
 	IssuerRef cmmeta.ObjectReference
 
 	// The PEM-encoded x509 certificate signing request to be submitted to the
-	// CA for signing.
+	// issuer for signing.
+	//
+	// If the CSR has a BasicConstraints extension, its isCA attribute must
+	// match the `isCA` value of this CertificateRequest.
+	// If the CSR has a KeyUsage extension, its key usages must match the
+	// key usages in the `usages` field of this CertificateRequest.
+	// If the CSR has a ExtendedKeyUsage extension, its extended key usages
+	// must match the extended key usages in the `usages` field of this
+	// CertificateRequest.
 	Request []byte
 
-	// IsCA will request to mark the certificate as valid for certificate signing
-	// when submitting to the issuer.
-	// This will automatically add the `cert sign` usage to the list of `usages`.
+	// Requested basic constraints isCA value.
+	//
+	// NOTE: If the CSR in the `Request` field has a BasicConstraints extension,
+	// it must have the same isCA value as specified here.
+	//
+	// If true, this will automatically add the `cert sign` usage to the list
+	// of requested `usages`.
 	IsCA bool
 
-	// Usages is the set of x509 usages that are requested for the certificate.
-	// Defaults to `digital signature` and `key encipherment` if not specified.
+	// Requested key usages/ extended key usages.
+	//
+	// NOTE: If the CSR in the `Request` field has a KeyUsage and/ or
```

Comment:
I think "and/or" hurts readability and doesn't add useful information. In the context of that sentence, it doesn't matter whether both keyusages and extendedusages are simultanously used. The important bit is that one of them at least is used. Thus, "or" is a better fit if the goal is clarity:

```suggestion
	// NOTE: If the CSR in the `Request` field uses the key usages
	// extension or the extended key usages extension, these extensions
	// must have the same values as specified here without any additional
	// values.
```